src/cargo/core/compiler/mod.rs

```diff
@@ -1135,6 +1140,13 @@ struct OutputOptions {
     /// of empty files are not created. If this is None, the output will not
     /// be cached (such as when replaying cached messages).
     cache_cell: Option<(PathBuf, LazyCell<File>)>,
+    /// If `true`, only process future incompat report messsages
+    /// (unless we see a malformed message), discarding all other
+    /// messages. This is used when replaying the output of units
+    /// that we don't want to show warnings for, so that
+    /// decreasing the verbosity level doesn't caues us to
+    /// incorrectly replay warnings.
+    replay_future_incompat_only: bool,
```

Comment:
I think this is probably better renamed as `show_warnings` like the original input for the option rather than making this future-incompat-specific. That's basically what this option is, which is that if we're unconditionally reading the output file of rustc then we just want to skip warnings, but other directives may wish to be processed.

Reply:
That's not what this flag does, though. If `report_future_incompat_only` is `true`, then we bail out of `on_stderr_line_inner` [if we fail to parse a future-incompat-report message](https://github.com/Aaron1011/cargo/blob/3199de8b5de6f440cdb6e7a3c8401f351afa7344/src/cargo/core/compiler/mod.rs#L1236-L1238), *before* attempting to do any other parsing.

I'm intentionally skipping the processing of other directives, because that's the current behavior of Cargo (we don't parse *any* directives, since we don't even read the file). We could put any future-incompat-report lines in their own file during recording of rustic output, but that seems much more complicated.

Reply:
The change in this PR is to unconditionally on every single build *always* parse all historical outputs of rustc. We can't change that. I would prefer to not have tons of future-incompat-report related things all over Cargo since it does not seem like a feature that justifies so much weight. By renaming this to `show_warnings` it means that it's just whether or not upstream compiler warnings are shown. Future-incompat messages and all other output from rustc is always processed as usual, where we hide future-incompat, jobserver, etc.

Reply:
My concern was that this could have other side effects, particularly if we ever add any other types of recorded messages. However, if you'd like to limit the amount of logic specific to future-incompat stuff, I'll change this to `show_warnings`.